script/zonemaster_backend_testagent

```diff
@@ -60,56 +61,13 @@ $opt_outfile //= '/var/log/zonemaster/zonemaster_backend_testagent.out';
 $loglevel    //= 'info';
 $loglevel = lc $loglevel;
 
-$loglevel =~ /^(?:trace|debug|info|inform|notice|warning|warn|error|err|critical|crit|fatal|alert|emergency)$/ or die "Error: Unrecognized --loglevel $loglevel\n";
-
-# Returns a Log::Dispatch object logging to STDOUT
-#
-# This procedure duplicates the STDOUT file descriptor to make sure that it keeps
-# logging to the same place even if STDOUT is later redirected.
-sub log_dispatcher_dup_stdout {
-    my $min_level = shift;
-
-    open( my $fd, '>&', \*STDOUT ) or die "Can't dup STDOUT: $!";
-    my $handle = IO::Handle->new_from_fd( $fd, "w" ) or die "Can't fdopen duplicated STDOUT: $!";
-    $handle->autoflush(1);
-
-    return Log::Dispatch->new(
-        outputs => [
-            [
-                'Handle',
-                handle    => $handle,
-                min_level => $min_level,
-                newline   => 1,
-                callbacks => sub {
-                    my %args = @_;
-                    $args{message} = sprintf "%s [%d] %s - %s", strftime( "%FT%TZ", gmtime ), $PID, uc $args{level}, $args{message};
-                },
-            ],
-        ]
-    );
-}
-
-# Returns a Log::Dispatch object logging to a file
-sub log_dispatcher_file {
-    my $min_level = shift;
-    my $log_file  = shift;
-
-    return Log::Dispatch->new(
-        outputs => [
-            [
-                'File',
-                filename  => $log_file,
-                mode      => '>>',
-                min_level => $min_level,
-                newline   => 1,
-                callbacks => sub {
-                    my %args = @_;
-                    $args{message} = sprintf "%s [%d] %s - %s", strftime( "%FT%TZ", gmtime ), $PID, uc $args{level}, $args{message};
-                },
-            ],
-        ]
-    );
-}
+Log::Any::Adapter->set(
+    '+Zonemaster::Backend::Log',
+    log_level => $loglevel,
+    json => $logjson,
+    file => $logfile,
+);
+$log->context->{pid} = $PID;
```

Comment:
Here we assign the PID of the master process, right? Will all workers be logged with the same pid?

Reply:
Indeed it was always the master process pid, fixed.